presentation/src/main/java/me/androidbox/presentation/navigation/NavigationGraph.kt

```diff
@@ -89,6 +89,10 @@ fun NavigationGraph(
             val agendaViewModel: AgendaViewModel = hiltViewModel()
             val agendaScreenState by agendaViewModel.agendaScreenState.collectAsStateWithLifecycle()
 
+            LaunchedEffect(key1 = agendaScreenState.agendaItems) {
+                agendaViewModel.fetchAgendaItems(agendaScreenState.selectedDate)
+            }
```

Comment:
What's the use of this?

When the agenda items change, you fetch agenda items again?
I'm not fully understanding

Reply:
Same here 🤔

Reply:
@MKiperszmid @philipplackner maybe I should have made a comment here. Basically, when the user chooses a different date I would like to fetch again to get those gender items based on that date. Is there a better way to do this?

Reply:
The thinking is correct, but not applied correctly. Your LaunchedEffect will be re-executed whenever `agendaItems` changes. But you want to re-execute it whenever `selectedDate` changes no? Otherwise picking a different date will not trigger this block.
But either way, I'd not work with LaunchedEffect for that, but rather just send an event to the ViewModel, like `OnDateSelected(newDate)` when a new date is submitted in the date picker dialog. Then the ViewModel's responsibility is to decide what happens in that case.

Reply:
@MKiperszmid @philipplackner I think I need this LaunchEffect block. The reason is that when the user adds another Event from the EventScreen and that event has been inserted. I need to trigger fetching all the events from the DB again to show the one that has just been inserted. 

Is there a better way to do this?

Reply:
Don't you just get a flow from the DB which updates the UI automatically?

Reply:
@philipplackner My insert is a suspend function but fetching is a flow:
    
    fun getEventsFromTimeStamp(startTimeStamp: Long, endTimeStamp: Long): Flow<ResponseState<List<Event>>>
    
    suspend fun insertEvent(event: Event): ResponseState<Unit>
    
    suspend fun deleteEventById(id: String): ResponseState<Unit>

Reply:
Insert should be a suspend

The Flow in fetch, will automatically detect when a new insert was done, and automatically retrieve the new information